electionleaflets/apps/leaflets/static/js/leaflet_uploader.js

```diff
@@ -0,0 +1,104 @@
+var getCookie = function(name) {
+  var cookieValue = null;
+  if (document.cookie && document.cookie !== '') {
+    var cookies = document.cookie.split(';');
+    for (var i = 0; i < cookies.length; i++) {
+      var cookie = cookies[i].trim();
+      // Does this cookie string begin with the name we want?
+      if (cookie.substring(0, name.length + 1) === (name + '=')) {
+        cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
+        break;
+      }
+    }
+  }
+  return cookieValue;
+};
+var handleError = function(err) {
+    console.debug(err)
+  var error = '<h2>Sorry, something has gone wrong</h2> <p>If the problem continues, ';
+  error += '<a href="mailto:pollingstations@democracyclub.org.uk">contact us</a>.</p>';
+  document.getElementById("error").innerHTML = error;
+  document.getElementById("error").hidden = false;
+  document.getElementById("submit").disabled = false;
+};
+var serializeFile = function(file) {
+  if (file == null) return null;
+  return {
+    name: file.name,
+    size: file.size,
+    type: file.type,
+  };
+};
+var getPresignedPostData = function() {
+  return new Promise(function(resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    var url = '{% url "file_upload_index" %}';
```

Comment:
Probably telling you something you already know (or will quickly discover when you try to run this), but this won't work in a js file. You'll need to move this tag to a template and either pass the value in somewhere or write it to the DOM in the template and read it back out here.

Also, if you'd like me to sod off and stop drive-by reviewing, let me know. I just got a pile of notifications from this repo yesterday so I started poking about making a nuisance of myself :smile:

Reply:
:D 

Thanks! This was actually a copy/paste from your work on the WDIV self serve upload stuff that I think you had inline in the template. As it happens, I have found and fixed this in a later commit...

I've also worked a bit on a local implementation of the pre-signed dance, allowing testing / use locally - will push in a bit.